R/run_eggla_gwas.R

```diff
@@ -586,61 +618,48 @@ run_eggla_gwas <- function(
           idcol = "trait_model"
         ),
         old = function(x) sub("^#", "", x)
-      )[TEST %in% "ADD" & !is.na(P), -c("TEST")]
+      )[
+        TEST %in% "ADD" & !is.na(P),
+        -c("TEST", "T_STAT", "OBS_CT")
+      ]
 
       output_results_file <- sprintf("%s.results.gz", results_file)
 
       if (!quiet) message(sprintf("[%s] Writing annotated results file ...", basename(vcf)))
-      if (use_info) {
-        output_dt <- data.table::merge.data.table(
-          y = results[j = A1 := NULL],
-          x = annot,
-          by.y = c("ID", "CHROM", "POS", "ALT", "REF"),
-          by.x = c("SNPID", "CHR", "POS", "EFFECT_ALLELE", "NON_EFFECT_ALLELE"),
-          all.y = TRUE
-        )
-        data.table::setcolorder(
-          x = output_dt,
-          neworder = c(
-            "trait_model",
-            "SNPID", "STRAND", "BUILD",
-            "CHR", "POS", "EFFECT_ALLELE", "NON_EFFECT_ALLELE",
-            "N", "N0", "N1", "N2", "EAF",
-            "BETA", "SE", "P",
-            "HWE_P", "CALL_RATE",
-            "IMPUTED", "INFO_TYPE", "INFO",
-            "T_STAT", "OBS_CT", "ERRCODE"
-          )
-        )
-      } else {
-        output_dt <- data.table::setnames(
-          x = results[j = A1 := NULL][
-            j = `:=`(
-              STRAND = strand,
-              BUILD = build
-            )
-          ],
-          old = c("ID", "CHROM", "POS", "ALT", "REF"),
-          new = c("SNPID", "CHR", "POS", "EFFECT_ALLELE", "NON_EFFECT_ALLELE")
-        )
-        data.table::setcolorder(
-          x = output_dt,
-          neworder = intersect(
-            x = c(
-              "trait_model",
-              "SNPID", "STRAND", "BUILD",
-              "CHR", "POS", "EFFECT_ALLELE", "NON_EFFECT_ALLELE",
-              "N", "N0", "N1", "N2", "EAF",
-              "BETA", "SE", "P",
-              "HWE_P", "CALL_RATE",
-              "IMPUTED", "INFO_TYPE", "INFO",
-              "T_STAT", "OBS_CT", "ERRCODE"
+      data.table::fwrite(
+        x = data.table::setnames(
+          x = Reduce(
+            f = function(x, y) merge(x, y, by = intersect(colnames(x), colnames(y))),
+            x = list(
+              data.table::fread(sprintf("%s.vmiss", results_file))[
+                j = list(
+                  `#CHROM`,
+                  ID,
+                  F_MISS,
+                  N = OBS_CT - MISSING_CT
+                )
+              ],
+              data.table::fread(sprintf("%s.afreq", results_file))[j = -c("OBS_CT")],
```

Comment:
@annihei The `--freq` PLINK2 option generate the `afreq` file which contains allele frequencies to be added in the output.

Reply:
> A: For N why don't we use `N = OBS_CT - MISSING_DOSAGE_CT` instead of MISSING_CT? `--missing vcols=+pos,+ref,+alt,+nmissdosage`

> M: Because, I did not know^^ Although, does it make a difference when we are using the `"dosage=DS"` specifier? Especially, since it basically says to PLINK to only look at dosage, thus in this case `MISSING_CT` should be the same as `MISSING_DOSAGE_CT`.

> A: I've been confusing with this as well.. See the row 4 :
> 
> > vmiss %>% filter(MISSING_CT > 0)
> > #CHROM      POS          ID REF ALT MISSING_DOSAGE_CT MISSING_CT OBS_CT
> > 1:     21  9414087  21:9414087   G   A                0          1   2903
> > 2:     21  9414249  21:9414249   T   C                 0          1   2903
> > 3:     21  9415411  21:9415411   A   T                 0          1   2903
> > 4:     21  9415943  21:9415943   G   A                 0      2871   2903
> > 5:     21  9418624  21:9418624   C   T                 0          1   2903

> A: Edit: there was a missing column in my output I think we use the F_MISS
> 
> > vmiss %>% filter(MISSING_CT > 0)
> > #CHROM      POS          ID REF ALT MISSING_DOSAGE_CT MISSING_CT OBS_CT      F_MISS
> > 1:     21  9414087  21:9414087   G   A                 0          1   2903 0.000344471
> > 2:     21  9414249  21:9414249   T   C                 0          1   2903 0.000344471
> > 3:     21  9415411  21:9415411   A   T                 0          1   2903 0.000344471
> > 4:     21  9415943  21:9415943   G   A                 0       2871   2903 0.988977000
> > 5:     21  9418624  21:9418624   C   T                 0          1   2903 0.000344471

Reply:
@annihei This is where the results from `--freq` is used (line 642).
I am using all columns but the `OBS_CT` from the output.

Reply:
OK yes so this is for the EAF?

Reply:
Yes, allele frequencies information.
Directly the output from PLINK2.